src/fides/api/api/v1/endpoints/privacy_request_endpoints.py

```diff
@@ -1805,80 +1790,65 @@ def get_individual_privacy_request_tasks(
 
 
 @router.post(
-    PRIVACY_REQUEST_TASK_QUEUE,
+    PRIVACY_REQUEST_REQUEUE,
     dependencies=[
         Security(verify_oauth_client, scopes=[PRIVACY_REQUEST_CALLBACK_RESUME])
     ],
-    response_model=Dict,
+    response_model=PrivacyRequestResponse,
 )
-def queue_task_manually(
+def requeue_privacy_request(
     privacy_request_id: str,
-    task_id: str,
     *,
     db: Session = Depends(deps.get_db),
-) -> Dict:
-    """
-    Manually queue a privacy request from a specific task if applicable
+) -> PrivacyRequestResponse:
     """
-    try:
-        privacy_request, request_task, _ = run_prerequisite_task_checks(
-            db, privacy_request_id, task_id
-        )
-    except (PrivacyRequestNotFound, RequestTaskNotFound) as exc:
-        raise HTTPException(
-            status_code=HTTP_404_NOT_FOUND,
-            detail=str(exc),
-        )
-    except UpstreamTasksNotReady as exc:
-        raise HTTPException(
-            status_code=HTTP_400_BAD_REQUEST,
-            detail=str(exc),
-        )
+    Endpoint for manually re-queuing a stuck Privacy Request from selected states - use with caution.
 
-    if privacy_request.status != PrivacyRequestStatus.in_processing:
-        raise HTTPException(
-            status_code=HTTP_400_BAD_REQUEST,
-            detail=f"Request failed. Cannot run {request_task.action_type.value} task {request_task.id} for privacy request {privacy_request.id} of status {privacy_request.status.value}",
-        )
+    Don't use this unless the Privacy Request is stuck.
+    """
+    pr: PrivacyRequest = get_privacy_request_or_error(db, privacy_request_id)
 
-    if request_task.status in COMPLETED_EXECUTION_LOG_STATUSES:
+    if pr.status not in [
+        PrivacyRequestStatus.pending,
+        PrivacyRequestStatus.approved,
+        PrivacyRequestStatus.in_processing,
+    ]:
         raise HTTPException(
             status_code=HTTP_400_BAD_REQUEST,
-            detail=f"Request failed. Cannot run {request_task.action_type.value} task {request_task.id} for privacy request {privacy_request.id} with status {request_task.status.value}",
+            detail=f"Request failed. Cannot re-queue privacy request {pr.id} with status {pr.status.value}",
         )
 
-    if request_task.action_type == ActionType.erasure and not all(
-        pr.status in COMPLETED_EXECUTION_LOG_STATUSES
-        for pr in privacy_request.access_tasks
-    ):
-        raise HTTPException(
-            status_code=HTTP_400_BAD_REQUEST,
-            detail=f"Request failed. Cannot run {request_task.action_type.value} task {request_task.id} when access tasks haven't completed.",
-        )
-    # TODO prevent re-running access task after erasure section has started?
+    # Both DSR 2.0 and 3.0 cache checkpoint details
+    checkpoint_details: Optional[
+        CheckpointActionRequired
+    ] = pr.get_failed_checkpoint_details()
+    resume_step = checkpoint_details.step if checkpoint_details else None
```

Comment:
Both DSR 2.0 and DSR 3.0 use this cached checkpoint if the Privacy Request fails early so we can resume from a specific checkpoint instead of running from the beginning.  

Additionally DSR 3.0 stores Request Tasks that also allow you to deduce at what checkpoint the Privacy Request failed, so that is also added here as a backup.  

An example of why we currently try to resume from checkpoints instead of the beginning: if an erasure request fails inside of the erasure graph, it is possible we've already erased some of the data that would make it difficult to re-run the access section.